src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -831,4 +873,155 @@ private static void SetGlobalSettings(
             logger?.SendMessage(TestMessageLevel.Warning, string.Format(CultureInfo.CurrentCulture, Resource.InvalidValue, disableParallelizationString, "DisableParallelization"));
         }
     }
+
+    private static void ParseBooleanSetting(IConfiguration configuration, string key, IMessageLogger? logger, Action<bool> setSetting)
+    {
+        if (configuration[$"mstest:{key}"] is string value)
+        {
+            if (bool.TryParse(value, out bool result))
+            {
+                setSetting(result);
+            }
+            else
+            {
+                logger?.SendMessage(TestMessageLevel.Warning, string.Format(CultureInfo.CurrentCulture, Resource.InvalidValue, value, key));
+            }
+        }
```

Comment:
Nit: I find it easier mentally when there is less indenting

```suggestion
        if (configuration[$"mstest:{key}"] is not string value)
        {
            return;
        }
        
        if (bool.TryParse(value, out bool result))
        {
            setSetting(result);
        }
        else
        {
            logger?.SendMessage(TestMessageLevel.Warning, string.Format(CultureInfo.CurrentCulture, Resource.InvalidValue, value, key));
        }
```